### ISSUE TITLE
Translations ? 

### ISSUE THREAD

**Issue by WoMec** [zivillian/ism7mqtt#148]
all entities in the home assistant are in German. is it possible to switch the language to english or another? 

![image](https://github.com/user-attachments/assets/81bf6623-32bc-44af-b9f2-beaab0df9cdc)

**Comment by framps** [comment]
ism7mqtt just uses the names exposed by your Wolf system and they are used in the published mqtt message. ism7mqtt just retrieves the data. The data presentation is done by HA. So it's HA who has to handle the translation into different languages. You should contact the HA development team and ask to handle the translation. Not sure whether they will accept this request.

**Comment by WoMec** [comment]
Thank you for the explanation. I understand that ism7mqtt retrieves data directly from the Wolf system, and the language comes from there. However, even when testing outside of Home Assistant, I notice that the data published to the MQTT server is still in German.

I was wondering if ism7mqtt uses some sort of mapping or lookup table to convert raw system data (e.g., variable identifiers and values) into the human-readable text published to MQTT? If such a table exists, would it be possible to modify it to provide translations, at least into English?

This could make the data more accessible for non-German speakers. I’d be happy to assist with creating a translation or testing this functionality if needed.

**Comment by framps** [comment]
> I was wondering if ism7mqtt uses some sort of mapping or lookup table to convert raw system data (e.g., variable identifiers and values) into the human-readable text published to MQTT? If such a table exists, would it be possible to modify it to provide translations, at least into English?

I see your point. I don't know whether Wolf equipment is sold outside of Germany. I frankly doubt and nobody will care about any translation on Wolf's side. So it's a question whether the ism7mqtt development team want's to spend any translation effort even Wolf is only used in Germany.

@WoMec What's you native language and in which country do you use Wolf equipment?

**Comment by WoMec** [comment]
I’m from the Czech Republic, where Wolf equipment is commonly sold and used. While the equipment might be primarily marketed in Germany, it’s not limited to that region. It would be great if some translation options could be considered to make the data more accessible for users in other countries as well.

**Comment by framps** [comment]
I didn't know Wolf also solds it's equipment in our country. So I agree it makes sense to think about to support translation of all the German terms into English and have some kind of language configuration option. It would be even much better to have translations into other languages Wolf's equipment is sold  - but that's challenging. But at leat English should be supported.

**Comment by WoMec** [comment]
Agree, English would be perfect and versatile for almost every user :)

**Comment by manison** [comment]
Some time ago I added parameter localization in my fork here: https://github.com/manison/ism7mqtt/tree/parameter-localization.  If interested I can create a PR.

**Comment by WoMec** [comment]
@manison that would be great :)

**Comment by LAdegeest** [comment]
I managed to install ism7mqtt in HA yesterday (many thanks to @b3nn0  ). Great improvement compared w the HA Wolflink-integration. Although not guaranteed, I can still run the Wolf Smartset app. There one can select any language and witch. At least it may help you to understand which parameter is what. But a buildin translation would be a great addition.

**Comment by zivillian** [comment]
I've implemented the translation for entity names in home assistant discovery and enum values. It would be great if someone could test the binaries.

The language can be selected via the new parameter `-l CZE` or `ISM7_LANGUAGE=CZE`.

The following languages are supported:
| Language | Code |
| --- | --- |
| German | DEU |
| Chinese simplified | CHN |
| Greek | GRC |
| Estonian | EST |
| Croatian | HRV |
| Latvian | LVA |
| Lithuanian | LTU |
| Romanian | ROU |
| Italian | ITA |
| Spanish | ESP |
| French | FRA |
| Polish | POL |
| Czech | CZE |
| Slovak | SVK |
| Russian | RUS |
| Danish | DNK |
| Hungarian | HUN |
| English | GBR |
| Turkish | TUR |
| Dutch | NLD |
| Bulgarian | BUL |
| Portuguese | POR |

**Comment by WoMec** [comment]
Hi,

I tested the Czech translation, and it is partially correct. The JSON keys are still in the original (German) language, but the JSON "text" values are translated correctly.

```json
{
  "Funktion des Bedienmoduls": {
    "value": 0,
    "text": "systém"
  },
  "Zeitprogramm Zirkulation Heizgerät": {
    "value": 2,
    "text": "2"
  },
  "Raumeinflussfaktor": 4,
  "Außenfühler gemittelt Berechnungszeitraum": 3,
  "Antilegionellenfunktion": {
    "value": 0,
    "text": "Vyp"
  },
  "Freigabe Parallelbetrieb": {
    "value": 0,
    "text": "Vyp"
  },
  "Raumtemperaturabhängige Abschaltung": {
    "value": 1,
    "text": "Zap"
  },
  "Absenkstopp": -16,
  "Warmwasserminimaltemperatur": 25,
  "PI-Regler Raumtemperatur Kp": 20,
  "PI-Regler Raumtemperatur Tn": 1,
  "Startzeit Antilegionellenfunktion": "18:00:00",
  "Uhrzeit": "22:51:00",
  "Datum": "07.01.2025 0:00:00",
  "Partymodus": {
    "value": 0,
    "text": "Deaktivován"
  },
  "Urlaubsmodus": {
    "value": 0,
    "text": "Deaktivován"
  },
  "Anforderung Lüftung": {
    "value": 8,
    "text": "Automatika VYP"
  },
  "1x Warmwasser": {
    "value": 0,
    "text": "Vyp"
  },
  "BUSCONFIG_Erkannte-Geraete-Low-Byte": 256,
  "BUSCONFIG_Erkannte-Geraete-High-Byte": 0,
  "BUSCONFIG_Softwareversion": 300,
  "BUSCONFIG_Sollwertkorrektur": 0,
  "SmartHome System": {
    "value": 0,
    "text": "žádné spojení"
  },
  "Freigabe SmartHome": {
    "value": 1,
    "text": "Zap"
  },
  "Außentemperatur": 1.1,
  "Wartungsmeldung": {
    "value": 1,
    "text": "V závislosti na provozu"
  },
  "Wartung Betriebsabhängig": 20,
  "Wartung Datumsabhängig Fälligkeitsdatum": "01.01.2064 0:00:00",
  "BUSCONFIG_Vereinfachter-Modus": 0,
  "Programmwahlschalter gemeinsam": {
    "value": 0,
    "text": "Vyp"
  },
  "Anpassung Raumfühler BM-2 0": 0,
  "Korrektur Außentemperatur": 0,
  "BUSCONFIG_SmartHome-System": 0,
  "BUSCONFIG_Programmwahlschalter-Gemeinsam": 0,
  "Firmware": 3,
  "Uhrzeit Ende": {
    "330000": "01:50:00",
    "330006": "01:50:00"
  },
  "Datum Ende": {
    "330001": "08.01.2025 0:00:00",
    "330007": "08.01.2025 0:00:00"
  },
  "Uhrzeit Start": {
    "330002": "22:50:00",
    "330008": "22:50:00"
  },
  "Datum Start": {
    "330003": "07.01.2025 0:00:00",
    "330009": "07.01.2025 0:00:00"
  },
  "Uhrzeit BM2": {
    "330004": "22:51:00",
    "330010": "22:51:00"
  },
  "Datum BM2": {
    "330005": "07.01.2025 0:00:00",
    "330011": "07.01.2025 0:00:00"
  }
}
```

**Comment by zivillian** [comment]
Thanks for testing. I didn't think about the json output and kept the keys untranslated to not change the MQTT topics. But you are right, that for json output it doesn't make sense to keep them in german.
I'm not yet sure if I should just translate everything and break existing integrations if someone changes the language.

**Comment by WoMec** [comment]
It's also possible to test translation in Home Assistant?

**Comment by b3nn0** [comment]
> Thanks for testing. I didn't think about the json output and kept the keys untranslated to not change the MQTT topics. But you are right, that for json output it doesn't make sense to keep them in german. I'm not yet sure if I should just translate everything and break existing integrations if someone changes the language. One additional issue is, that the translations are not idempotent - so it get's even worse to revert the translation when sending values.

For HA discovery topics, at least the unique_id and object_id in the discovery info should remain the same so changing language doesn't duplicate the individual sensors etc.

**Comment by b3nn0** [comment]
Once there is a docker image released with this by @zivillian , I will update the experimental addon to have a language setting.

**Comment by zivillian** [comment]
I've merged the feature and pushed a new image to the [master tag](https://hub.docker.com/layers/zivillian/ism7mqtt/master/images/sha256-aa77bc72c2578da222a89cb6a35c7e0a96449b188ed1b0527d642b3059bac275).

**Comment by b3nn0** [comment]
I pushed the experimental addon and it seems to do something:
![image](https://github.com/user-attachments/assets/aeb40807-a6c6-42f9-a6c4-11a0cb7583ee)

For those who want to test: I think it should work if you go to the addon page in HA and hit the "Rebuild" button. Maybe clear some browser caches or something, then you should see a language setting.
![image](https://github.com/user-attachments/assets/5e59ce89-d6f7-4f89-bb8e-d7a0e4edcf54)

**Comment by WoMec** [comment]
translation in HA looks flawless to me :) good work @b3nn0 @zivillian 

![Image](https://github.com/user-attachments/assets/6818991f-c8f3-41d8-8023-50a2d0e6b614)
